Modify config so SHREDDIT_BEFORE supports duration notation

BootsSiR opened the pull request:
Allows you to specify a duration notation (ie: '-30d', '-2w', '-5h') or a date in the SHREDDIT_BEFORE field.  This enables you to run this at any time and have it delete any comments more than say 30 days old (-30d)

andrewbanchich commented on src/cli.rs:
```diff
@@ -1,9 +1,28 @@
+/// Parses `SHREDDIT_BEFORE` to support:
+/// - Absolute timestamps (ISO 8601) → `"2025-01-31T03:16:30Z"`
+/// - Negative durations (`-30d`, `-2w`, `-5h`) → Converts to `Utc::now() - duration`
+fn parse_before(input: &str) -> Result<DateTime<Utc>, String> {
```
nice! thanks so much for the contribution :smile:

BootsSiR replied:
ok, moved to the library you suggested.

BootsSiR commented on Cargo.toml:
```diff
@@ -25,3 +25,4 @@ serde_json = "1.0.139"
 tokio = { version = "1.43.0", features = ["rt", "macros"] }
 tracing = "0.1.37"
 tracing-subscriber = { version = "0.3.19", features = ["fmt", "env-filter"] }
+parse_duration = "2.1.1"
```
ah, I never even checked that.  I'll cut the code over to your suggestion tomorrow.

andrewbanchich replied:
thanks!

BootsSiR replied:
moved to the library you suggested.  Seems to work well, just have to use the long form instead of short - ie: -30d would be -30 days.  Seems like a more flexible parser though.

andrewbanchich replied:
awesome, thank you!

BootsSiR replied:
I just pushed a change to fix the formatting errors that were failing as well.

andrewbanchich commented:
hi @BootsSiR , CI was failing because of formatting so i pulled and noticed a clippy warning. i fixed that, then added a unit test, which led me to do some refactors. i updated the PR with what i _think_ seems correct. hope you don't mind!

let me know if this is good / what you had in mind / works!

BootsSiR commented:
> hi @BootsSiR , CI was failing because of formatting so i pulled and noticed a clippy warning. i fixed that, then added a unit test, which led me to do some refactors. i updated the PR with what i _think_ seems correct. hope you don't mind!
> 
> let me know if this is good / what you had in mind / works!

Rust isn't my native language so works for me 😁

andrewbanchich commented:
merged! thank you @BootsSiR
